x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/analysis/Analyzer.java

```diff
@@ -1067,24 +1068,19 @@ public static Expression castStringLiteral(Expression from, DataType target) {
                     target,
                     e.getMessage()
                 );
-                return new UnsupportedAttribute(
-                    from.source(),
-                    String.valueOf(from.fold()),
-                    new UnsupportedEsField(String.valueOf(from.fold()), from.dataType().typeName()),
-                    message
-                );
```

Comment:
Unrelated, but maybe it's better not to fake an ES field here, `UnresolvedAttribute` seems to do the trick as well.